x/ecocredit/params.go

```diff
@@ -37,12 +45,45 @@ func validateCreditClassFee(i interface{}) error {
 	return nil
 }
 
-func NewParams(creditClassFee sdk.Coins) Params {
+func validateCreditTypes(i interface{}) error {
```

Comment:
should we enforce the units here? was wondering if we should have a hard coded list or something of valid unit types like kg, tonne, etc

Reply:
not a bad idea, but then we'd need a units library... something like UCUM might work

Reply:
I found one golang UCUM library: https://github.com/BertVerhees/ucum. It's old and not maintained. We'd probably just want to download the XML file from https://ucum.org/trac and parse that.

@blushi have you thought much about units from the registry SHACL perspective?

A good alternative would be http://qudt.org which is more up to date, but requires us parsing RDF files...

Reply:
in short, probably a good idea but maybe we address later..?

Reply:
ok ill open an issue for this

Reply:
On the registry, I've been using http://qudt.org indeed in SHACL schemas